core/utils/ethabi.go

```diff
@@ -26,21 +27,31 @@ const (
 	FormatBool = "bool"
 )
 
-// GenericEncode eth encodes values based on the provided types
-func GenericEncode(types []string, values ...interface{}) ([]byte, error) {
-	if len(values) != len(types) {
-		return nil, errors.New("must include same number of values as types")
+// ABIEncode is the equivalent of abi.encode.
+// See a full set of examples https://github.com/ethereum/go-ethereum/blob/420b78659bef661a83c5c442121b13f13288c09f/accounts/abi/packing_test.go#L31
+func ABIEncode(abiStr string, values ...interface{}) ([]byte, error) {
```

Comment:
Passing in an ABI string seems less user-friendly than the former version of this method. Is this safer / more all-encompassing than the previous implementation?

Reply:
yeah the main thing is that the previous approach didn't support nesting. Passing in the ABI makes it universal. Will error if the string is invalid or doesn't match values